src/main/resources/fileTemplates/testMeIncludes/TestMe macros.java.ft

```diff
@@ -106,4 +106,85 @@ assertEquals(result#{if}($TestSubjectUtils.isJavaFuture($method.returnType)).get
 #end
 #end
 #end
+#end
+
+##----------------------------------------------------------------------------------------------------------------------
+## PowerMock macros
+##----------------------------------------------------------------------------------------------------------------------
+
+##----------------------------------------------------------------------------------------------------------------------
+## renderInternalMethodCallsStubs($methods,$testedClass)
+## render internal method call stubs (currently PowerMock used)
+## Param: $method - the tested method, $testedClass - the tested class
+##----------------------------------------------------------------------------------------------------------------------
+#macro(renderInternalMethodCallsStubs $method $testedClass)
+    $testedClass.canonicalName $StringUtils.deCapitalizeFirstLetter($testedClass.name)Spy = spy($StringUtils.deCapitalizeFirstLetter($testedClass.name));
+    #foreach($methodCall in $method.directMethodCalls)
+        #if($TestSubjectUtils.isMethodOwnedByClass($methodCall.method, $testedClass) && $methodCall.method.hasReturn())
+            #if($methodCall.method.hasParams())
+                doReturn($TestBuilder.renderReturnParam($methodCall.method,$methodCall.method.returnType,"${methodCall.method.name}Response",$replacementTypes,$defaultTypeValues)).when(${StringUtils.deCapitalizeFirstLetter($testedClass.name)}Spy, "${methodCall.method.name}", $PowerMockBuilder.buildMockArgsMatchers(${methodCall.method.methodParams},"Java"));
+            #else
+                doReturn($TestBuilder.renderReturnParam($methodCall.method,$methodCall.method.returnType,"${methodCall.method.name}Response",$replacementTypes,$defaultTypeValues)).when(${StringUtils.deCapitalizeFirstLetter($testedClass.name)}Spy, "${methodCall.method.name}");
+            #end
+        #end
+    #end
+#end
+
+##----------------------------------------------------------------------------------------------------------------------
+## renderMethodCallWithSpy($method,$testedClassName)
+## render method call use spy, currently PowerMock used
+## Param: $method - the tested method, $testedClassName - the tested class name
+##----------------------------------------------------------------------------------------------------------------------
+#macro(renderMethodCallWithSpy $method,$testedClassName)
+    #renderJavaReturnVar($method.returnType)#if($method.static)$testedClassName#{else}$StringUtils.deCapitalizeFirstLetter($testedClassName)Spy#end.${method.name}($TestBuilder.renderMethodParams($method,$replacementTypes,$defaultTypeValues));##
+#end
+
+
+##----------------------------------------------------------------------------------------------------------------------
```

Comment:
There's too much duplicated code in these 3 macros:
_renderMockedFields_
_renderMockStubs_
_renderMockVerifies_

Try the following steps:
- Extract  a common MockBuilder interface with common methods being used
- Add to top macro declarations: `#set($mockBuilder = $MockitoMockBuilder)` (add also `@vtlvariable` declaration for new global var `mockBuilder` - mapped to new interface)
- Replace current references of `$MockitoMockBuilder` in macro to new `$mockBuilder` global var
- In PowerMock template, after macro imported, add -  `#set($mockBuilder = $PowerMockBuilder)` . I'm assuming this will override the global var by the time macro function will be called